Error in generated c++ source code when passing sinks to functions

```
# file: foo.spicy

module foo;

function fun(s: sink): bool {
    s.close();
    return True;
}

type Y = unit {
    sink s;
    x: X(self.s);
};


type X = unit (s: sink){
    data: uint8;
    var a: bool;

    on data {
      self.a = fun(s);
    }
};
```

Causes the following error:

```
$ spicyc -j ./foo.spicy
/private/var/folders/xg/y2l9jptj6bq6gm7rz1mm2lc80000gn/T/foo_93d27a28ebfe11ba-92d90ada2067c1b3.cc:354:5: error: 'this' argument to member function 'close' has type 'const spicy::rt::Sink', but function is not marked const
    (*s).close();
    ^~~~
/Users/johanna/zeek/install/include/spicy/rt/sink.h:170:10: note: 'close' declared here
    void close() { _close(true); }
         ^
1 error generated.
[error] spicyc: JIT compilation failed
```

It can be resolved by either calling ``s.close()`` in ``on data``, or by changing the function signature to use ``inout`` for the sink. However, this still seems like a bug/inconsistency since one does not need ``inout`` when passing sinks to other units.

--- a/hilti/toolchain/src/compiler/codegen/types.cc
+++ b/hilti/toolchain/src/compiler/codegen/types.cc
@@ -653,7 +653,7 @@ struct VisitorStorage : hilti::visitor::PreOrder<CxxTypes, VisitorStorage> {
         else
             t = "*";
 
-        return CxxTypes{.base_type = t, .param_in = fmt("const %s", t), .param_inout = fmt("%s", t)};
+        return CxxTypes{.base_type = t, .param_in = fmt("%s", t), .param_inout = fmt("%s", t)};
     }
 
     result_t operator()(const type::stream::View& n) { return CxxTypes{.base_type = "::hilti::rt::stream::View"}; }

